process/block/baseProcess.go

```diff
@@ -111,18 +113,32 @@ func (bp *baseProcessor) checkBlockValidity(
 		return process.ErrWrongNonceInBlock
 	}
 
-	if headerHandler.GetNonce() != chainHandler.GetCurrentBlockHeader().GetNonce()+1 {
+	if currentBlockHeader.GetRound() >= headerHandler.GetRound() {
```

Comment:
can you keep same order of checked parameters in all conditionals? 
e.g for both verify currendBlockHeader.X (conditional) headerHandler.X
or the other way around

Reply:
Done